### PR TITLE
Demos C++ fix warnings

### PR DESCRIPTION
**Pull request by Wovchena**
Ignore `object_detection_demo_ssd_async` because there are still older IRs for the demo.

### CODE REVIEW

File: demos/interactive_face_detection_demo/detectors.cpp [hunk 1]
```diff
@@ -609,25 +508,8 @@ CNNNetwork EmotionsDetection::read(const InferenceEngine::Core& ie) {
 
     DataPtr emotionsOutput = outputInfo.begin()->second;
```
Comment (IRDonch) [inline demos/interactive_face_detection_demo/detectors.cpp]: Looks like this is unused now.

File: demos/interactive_face_detection_demo/detectors.cpp [hunk 2]
```diff
@@ -720,30 +602,17 @@ CNNNetwork FacialLandmarksDetection::read(const InferenceEngine::Core& ie) {
...
+    if (outName != outputFacialLandmarksBlobName) {
+        throw std::logic_error("Facial Landmarks Estimation network output layer unknown: " + outName
+                               + ", should be " + outputFacialLandmarksBlobName);
+    }
+    Data& data = *outputInfo.begin()->second;
+    data.setPrecision(Precision::FP32);
+    const SizeVector& outSizeVector = data.getTensorDesc().getDims();
+    if (outSizeVector.size() != 2 && outSizeVector.back() != 70) {
+        throw std::logic_error("Facial Landmarks Estimation network output layer should have 2 dimensions and 70 as"
+                               " a last dimension");
```
Comment (IRDonch) [inline demos/interactive_face_detection_demo/detectors.cpp]:
```suggestion
                               " the last dimension");
```

File: demos/multi_channel/object_detection_demo_yolov3/main.cpp [hunk 3]
```diff
@@ -298,7 +298,7 @@ std::map<std::string, YoloParams> GetYoloParams(const std::vector<std::string>&
                 }
             }
         } else {
-            params = network.getLayerByName(output_name.c_str());
+            throw std::runtime_error("Cant get ngraph::Function. Isn't the IR of version 10 provided?");
```
Comment (IRDonch) [inline demos/multi_channel/object_detection_demo_yolov3/main.cpp]:
```suggestion
            throw std::runtime_error("Can't get ngraph::Function. Make sure the provided model is in IR version 10 or greater.");
```

File: demos/object_detection_demo_faster_rcnn/main.cpp [hunk 4]
```diff
@@ -74,10 +72,6 @@ int main(int argc, char *argv[]) {
         slog::info << "Loading Inference Engine" << slog::endl;
         Core ie;
 
-        if (FLAGS_p_msg) {
```
Comment (IRDonch) [inline demos/object_detection_demo_faster_rcnn/main.cpp]: You should also remove the flag itself.

File: demos/interactive_face_detection_demo/detectors.cpp (hunk 1 above)
Comment (Wovchena) [reply demos/interactive_face_detection_demo/detectors.cpp]: removed

File: demos/object_detection_demo_faster_rcnn/main.cpp (hunk 4 above)
Comment (Wovchena) [reply demos/object_detection_demo_faster_rcnn/main.cpp]: removed

**Pull request merged**